### PR TITLE
feat: adding in etcd as iterable

### LINKED ISSUE #1227: keyv.iterator is not a function when using KeyvEtcd
keyv.iterator is not a function when using KeyvEtcd.

```ts
const keyv = new Keyv(new KeyvEtcd('etcd://localhost:2379'));
  await keyv.set('key1', 'value1');
  await keyv.set('key2', 'value2');

  console.log(keyv.iterator); // TypeError: keyv.iterator(...) is not a function
  for await (const [key, value] of keyv.iterator()) {
    console.log(key, value);
  }
```

In fact, KeyvEtcd has an iterator.
https://github.com/jaredwray/keyv/blob/27c49a00c7c0b12875eec48b071cf297152644bf/packages/etcd/src/index.ts#L126-L136

### PR DESCRIPTION
**Pull request by MAXLZ1**
**Please check if the PR fulfills these requirements**
- [x] Followed the [Contributing](https://github.com/jaredwray/keyv/blob/main/CONTRIBUTING.md) guidelines.
- [x] Tests for the changes have been added (for bug fixes/features) with 100% code coverage.
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
close #1227 

To make the test pass, I modified the package.json of the etcd package so that keyv points to the package in the workspace.

### CODE REVIEW

**Comment by jaredwray** [comment]
@MAXLZ1 - looks like there are some errors that need to get resolved. Also, did you add code for etcd to do iteration?

File: packages/etcd/src/index.ts [hunk 1]
```diff
@@ -42,6 +43,7 @@ export class KeyvEtcd<Value = any> extends EventEmitter {
 			url: '127.0.0.1:2379',
 			...url,
 			...options,
+			dialect: 'etcd',
```
Comment (MAXLZ1) [inline packages/etcd/src/index.ts:46]: I added the dialect: 'etcd' option when initializing KeyvEtcd.

Comment (MAXLZ1) [inline packages/etcd/src/index.ts:46]: I added the dialect: 'etcd' option when initializing KeyvEtcd.

File: packages/keyv/src/index.ts [hunk 2]
```diff
@@ -89,6 +89,7 @@ const iterableAdapters = [
 	'mongo',
 	'redis',
 	'valkey',
+	'etcd',
```
Comment (MAXLZ1) [inline packages/keyv/src/index.ts:92]: Add etcd to iterableAdapters.

**Comment by jaredwray** [comment]
@MAXLZ1 - thanks so much and we should have this update roll out in the next couple weeks.

**Pull request merged**